### DISCUSSION TITLE
Blank dictionary generated for deployment

### DISCUSSION THREAD

**Discussion by teddylambert** [nasa/fprime#1810] [Q&A] [answered]
I am working on a project that is using F-Prime 2.0.1 and am running into an issue where the generated `<>TopologyAppDictionary.xml` file is completely empty. I can purge, generate, build, and run the deployment fine, but the generate/build steps only ever create empty dictionary files. I looked through the steps in #1450 and confirmed that my naming is consistent. Additionally, the exact same files on another person's machine will generate the dictionary correctly. I also confirmed that both machines are using the same version of python and fprime-tools.

I know it's generally hard to debug "It doesn't work for me specifically errors", but is there anything I'm missing that could be causing the issue?

**Comment by LeStarch** [comment, marked as answer]
What OS are you on?  It could be file system case-sensitivity issues.  Also confirm that you have the same versions of python packages via:

```
pip freeze
```

This could see if something has diverged between computers.

**Reply by teddylambert** [reply]
I tried `pip freeze` with a colleague and confirmed packages were exactly the same but I was still having the issue.

I'm running on a Linux VirtualBox VM (Ubuntu Focal) which was configured using Vagrant. Colleagues' machines were set up in the same way, which was supposed to help with this "doesn't work for me issue" but I must've broken something on my own setup somewhere.

**Reply by SterlingPeet** [reply]
Thats definitely odd.  We want to preserve the current setup for debug comparison.  If you try to build a fresh vagrant machine from an adjacent folder using a copy of the Vagrantfile, do you get the same result?

**Reply by teddylambert** [reply]
Created a fully new machine, and looks like the dict is being generated now!
